src/include/duckdb/planner/operator_extension.hpp

```diff
@@ -0,0 +1,45 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/planner/operator_extension.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/common/common.hpp"
+#include "duckdb/execution/physical_plan_generator.hpp"
+#include "duckdb/planner/binder.hpp"
+
+namespace duckdb {
+
+// forward declaration to break circular inclusion
+class LogicalExtensionOperator;
+
+//! The OperatorExtensionInfo holds static information relevant to the operator extension
+struct OperatorExtensionInfo {
+	DUCKDB_API virtual ~OperatorExtensionInfo() {
+	}
+};
+
+typedef unique_ptr<PhysicalOperator> (*create_plan_function_t)(ClientContext &context, PhysicalPlanGenerator &generator,
+                                                               LogicalExtensionOperator &plan);
+typedef BoundStatement (*bind_function_t)(ClientContext &context, Binder &binder, OperatorExtensionInfo *info,
+                                          SQLStatement &statement);
+
+class OperatorExtension {
+public:
+	//! Creates a plan for a new logical operator injected into duckdb via the OperatorExtension's bind function or
+	//  the optimizer extension
+	create_plan_function_t CreatePlan;
```

Comment:
Should this perhaps be implemented as virtual methods on the `LogicalOperator` that are only used for the extension operators? It seems a bit tricky to know which extension knows how to handle which logical operator otherwise.